funcake_dags/field_counter.py

```diff
@@ -38,12 +38,14 @@ def __init__(self):
     def count(self, xml_string):
         parser = XMLPullParser(huge_tree=True)
         parser.feed(xml_string)
+        tags = ["dc", "record"]
         for event, elem in parser.read_events():
 
             elem_tag = QName(elem.tag).localname
-            if elem_tag == "record":
+            if elem_tag in tags:
+                if len(tags) == 2:
+                    tags = [elem_tag]
```

Comment:
@quoideneuf I'm confused by this statement.  Isn't `len(tags)` always equal to 2 by definition `tags = ["dc", "record"]`?

Reply:
@dkinzer the intention is to have counter be indifferent to whether it counts "record" or "dc", but only to choose one and ignore the other once it has chosen. I realize it is not a great solution but I felt like trying to be more logical could break something else.

Reply:
@quoideneuf I'm still not following.  len(tags) is always == 2 so why are you bothering with the if statement? Or am I missing something?

Reply:
@dkinzer It starts out with a length of 2 when it begins parsing the chunk of xml. But as soon as one or the other show up in the stream ("record" or "dc") it resets the `tags` to a single item array with the winning tag name. Sorry, this could probably be written more clearly...

Reply:
OK, I get it now. We are always forcing tag to be 1 (either dc or record but not both) ...hmm... there was a comment about ignoring dc explicitly ... hmm... should we just set tag to "record" always?  I'l merge and hope for the best.